### PR TITLE
Create User Page

### PR DESCRIPTION
**Pull request by anay-208**
# Create User Page

## Summary
Me & @Arinji2 have created a user page, which would display the users:
- Recently Marked Answers
- Join Date(A new column is created in users)
- Total Answers
- Modified Ui in posts
Screenshot of page:
User Page
Post Page

## Note
- **Database has been Restructured**:  Added `joinedAt`, and modified index to be sorted. Migration needs to be ran.
- **New Database Value**: In Sync user, I've added the logic to sync joinedAt, please also note that it'll be empty once the database is migrated.

## Changes Needed
- **Check the performance of leaderboard query**: The performance of leaderboard query needs to be checked, and query needs to be optimised. CC: @rafaelalmeidatk

### CODE REVIEW

**Comment by Arinji2** [comment]
With the latest commit, the Changes to UI are completed.

**Comment by anay-208** [comment]
I also checked that if the `joinedAt` column is empty, it'll just not show the value in profile page, until next user sync.

Please also verify it on your end

File: apps/web/app/user/[discordID]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+import { HeartIcon } from '@/components/icons/heart'
+import { MedalIcon } from '@/components/icons/medal'
+import { Post } from '@/components/post'
+import { getCanonicalUserUrl } from '@/utils/urls'
+import { Metadata } from 'next'
+import { db, sql } from '@nextjs-forum/db/node'
+import { notFound } from 'next/navigation'
+const getLeaderboardPosition = async (discordID: string) => {
+  const result = await db
+    .with('rankedUsers', (db) =>
```
Comment (rafaelalmeidatk) [inline apps/web/app/user/[discordID]/page.tsx:11]: This query works but it is reading every row from the table. I tested the following query and the execution plan seems to be better:
```sql
WITH userAnswers AS
  (SELECT "answersCount"
   FROM users
   WHERE "snowflakeId" = $1
)
SELECT 1 +
  (SELECT COUNT(*)
   FROM users
   WHERE "answersCount" >
       (SELECT "answersCount"
        FROM userAnswers)) AS rank;
```

File: apps/web/app/user/[discordID]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    ])
+    .execute()
+
+  return detailedPosts
+}
+
+export const generateMetadata = async ({
+  params,
+}: UserProps): Promise<Metadata> => {
+  // const userData = await getUserData(params.discordID)
```
Comment (rafaelalmeidatk) [inline apps/web/app/user/[discordID]/page.tsx:96]: Unused code

File: packages/db/migrations/6-create-profile-page.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    .createIndex('users_answersCount_desc_snowflakeId_desc_idx')
+    .on('users')
+    .columns(['answersCount desc', 'snowflakeId desc'])
+    .execute()
+}
+
+export async function down(db: Kysely<any>): Promise<void> {
+  await db.schema.alterTable('users').dropColumn('joinedAt').execute()
+
+  await db.schema.dropIndex('users_answersCount_idx').execute()
```
Comment (rafaelalmeidatk) [inline packages/db/migrations/6-create-profile-page.ts:19]: This is the wrong index name you created on the `up` function

**Review by rafaelalmeidatk — CHANGES_REQUESTED** [review]
The page looks great! I left some small suggestions

**Comment by anay-208** [comment]
@rafaelalmeidatk I also checked the query, and found out if 2 users have the same answerCount, Their rank will be same.

This is completely fine, but the thing is that, it is not how it works in the home page

**Comment by rafaelalmeidatk** [comment]
> @rafaelalmeidatk I also checked the query, and found out if 2 users have the same answerCount, Their rank will be same.
> 
> This is completely fine, but the thing is that, it is not how it works in the home page

@anay-208 uh yeah that makes sense, we can go back to using `RANK` then and I will monitor the query in the dashboard to make sure it's not using too many resources but it should be fine since we don't update the page very often

**Comment by anay-208** [comment]
@rafaelalmeidatk Done

**Review by rafaelalmeidatk — APPROVED** [review]
Thank you, this is awesome 🙏

**Pull request merged**